### ISSUE TITLE
Cattle IP Setup

### ISSUE THREAD

**Issue by gkawamoto** [itzg/dockerfiles#140]
I am using Rancher + Cattle in our cluster (http://rancher.com/) and we've found an issue regarding the way Cattle register the container at the network.
It seems that the container only joins the network after he's up and, consequently, after the container has run the `discoverAllGlobalIps` function in the start sequence.

To fix it in our setup I've changed the implementation to ignore certain subnets and wait so the right network is set. Something like this:

```bash
discoverAllGlobalIps() {
  if [ ${#IGNORE_NETWORK} -eq 0 ] 
  then
    IGNORE_NETWORK='999.999.999.999'
  fi
  printf "Finding IPs"
  while [ ${#ips} -eq 0 ]
  do
    printf "."
    ips=`ipaddr show scope global| grep -v "inet ${IGNORE_NETWORK}" | awk '$1 == "inet" { if (!match($2,"/32")) { gsub("/.*","",$2) ; addrs[length(addrs)] = $2 } } END { for (i in addrs) { if (i>0) printf "," ; printf addrs[i] } }'`
    sleep 1
  done
  echo " found! $ips"
  OPTS="$OPTS -E network.host=$ips"
}
```

It doesn't seem to affect the implementation for the Swarm, only slows a bit the startup for Cattle. Do you guys prefer a PR?

Thanks!

**Comment by itzg** [comment]
Thanks for investigating this and cross verifying in Swarm. Yes, a PR would be fantastic.

**Comment by itzg** [comment]
Fixed by #141
